internal/cli/internal/command/open/open.go

```diff
@@ -0,0 +1,69 @@
+package open
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/skratchdot/open-golang/open"
+	"github.com/spf13/cobra"
+
+	"github.com/superfly/flyctl/internal/buildinfo"
+	"github.com/superfly/flyctl/internal/cli/internal/app"
+	"github.com/superfly/flyctl/internal/cli/internal/command"
+	"github.com/superfly/flyctl/internal/cli/internal/flag"
+	"github.com/superfly/flyctl/internal/client"
+	"github.com/superfly/flyctl/pkg/iostreams"
+)
+
+func New() *cobra.Command {
+	const (
+		long = `Open browser to current deployed application. If an optional path is specified, this is appended to the
+		URL for deployed application
+`
+		short = "Open browser to current deployed application"
+
+		usage = "open [PATH]"
+	)
+
+	cmd := command.New(usage, short, long, run, command.RequireSession, command.RequireAppName)
+
+	cmd.Args = cobra.MaximumNArgs(1)
+
+	flag.Add(cmd,
+		flag.App(),
+		flag.AppConfig(),
+	)
+
+	return cmd
+}
+
+func run(ctx context.Context) error {
+	var (
+		path = "/"
+
+		appName = app.NameFromContext(ctx)
+	)
+
+	path = flag.FirstArg(ctx)
+
+	app, err := client.FromContext(ctx).API().GetApp(ctx, appName)
+	if err != nil {
+		return fmt.Errorf("failed retrieving app: %w", err)
+	}
+
+	if !app.Deployed {
+		return fmt.Errorf(`app has not been deployed yet. Try running "` + buildinfo.Name() + ` deploy --image flyio/hellofly"`)
```

Comment:
instead of 
```go
return fmt.Errorf(`app has not been deployed yet. Try running "` + buildinfo.Name() + ` deploy --image flyio/hellofly"`)
```

try 

```go
return fmt.Errorf(`app has not been deployed yet. Try running "%s" deploy --image flyio/hellofly"`, buildinfo.Name())
```

yes?

Reply:
It doesn't have work if you haven't run the flyctl launch. Plus it doesn't even make sense to suggest deploying the `flyio/hellofly` image when the user might be working with something different

Reply:
It doesn't have work if you haven't run the flyctl launch. Plus it doesn't even make sense to suggest deploying the `flyio/hellofly` image when the user might be working with something different

Reply:
> It doesn't have work if you haven't run the flyctl launch. Plus it doesn't even make sense to suggest deploying the `flyio/hellofly` image when the user might be working with something different. `Try running flyctl deploy` makes more sense to me as a user.

Then maybe prompt the user to launch the application first instead? We don't need to be specific, just give them a rough idea why this isn't working, imo.

Reply:
something like 

> "app has not been deployed yet. Please try deploying your app first"